ci2/ci/github.py

```diff
@@ -257,7 +259,8 @@ def batch_was_cancelled(batch):
             if len(batches) > 0:
                 self.batch = batches[0]
             elif run:
-                await self.start_build(batch)
+                async with repos_lock:
+                    await self.start_build(batch_client)
 
         if self.batch:
             seen_batch_ids.add(self.batch.id)
```

Comment:
down here we change build_state but don't set batch_changed, that seems wrong

Reply:
if a build is now success, we might want to merge it, right? batch_changed will trigger the watched branch heal loop that called this PR heal to try healing again and maybe merging.

Reply:
Good catch, you're right. Fixed.